### PR TITLE
[#5894] feat(iceberg): support Azure account key credential

### PR DESCRIPTION
**Pull request by orenccl**
### What changes were proposed in this pull request?

Support Azure account key credential

### Why are the changes needed?

Fix: #5894

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit Test IcebergRESTADLSAccountKeyIT at iceberg 1.6.0

### CODE REVIEW

File: api/src/main/java/org/apache/gravitino/credential/ADLSAccountKeyCredential.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+package org.apache.gravitino.credential;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+
+/** ADLS account key credential. */
+public class ADLSAccountKeyCredential implements Credential {
```
Comment (FANNG1) [inline api/src/main/java/org/apache/gravitino/credential/ADLSAccountKeyCredential.java:28]: How about using the name `AzureAccountKeyCredential`?  This is more general, if we support Azure blob storage, we could reuse it.

**Comment by FANNG1** [comment]
LGTM, just one naming issue.

File: api/src/main/java/org/apache/gravitino/credential/ADLSAccountKeyCredential.java (hunk 1 above)
Comment (orenccl) [reply api/src/main/java/org/apache/gravitino/credential/ADLSAccountKeyCredential.java]: Yes, I think it's better.

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTADLSAccountKeyIT.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import org.apache.gravitino.iceberg.common.IcebergConfig;
+import org.apache.gravitino.integration.test.util.BaseIT;
+import org.apache.gravitino.integration.test.util.DownloaderUtils;
+import org.apache.gravitino.integration.test.util.ITUtils;
+import org.apache.gravitino.storage.AzureProperties;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+@SuppressWarnings("FormatStringAnnotation")
+@EnabledIfEnvironmentVariable(named = "GRAVITINO_TEST_CLOUD_IT", matches = "true")
+public class IcebergRESTADLSAccountKeyIT extends IcebergRESTJdbcCatalogIT {
```
Comment (FANNG1) [inline iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTADLSAccountKeyIT.java:36]: Could you also rename this class and related content?

File: bundles/azure-bundle/src/main/java/org/apache/gravitino/abs/credential/ADLSAccountKeyProvider.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.Map;
+import org.apache.gravitino.credential.AzureAccountKeyCredential;
+import org.apache.gravitino.credential.Credential;
+import org.apache.gravitino.credential.CredentialConstants;
+import org.apache.gravitino.credential.CredentialContext;
+import org.apache.gravitino.credential.CredentialProvider;
+import org.apache.gravitino.credential.config.ADLSCredentialConfig;
+
+/** Generates ADLS account key to access ADLS data. */
+public class ADLSAccountKeyProvider implements CredentialProvider {
```
Comment (FANNG1) [inline bundles/azure-bundle/src/main/java/org/apache/gravitino/abs/credential/ADLSAccountKeyProvider.java:31]: could you also rename this class?

File: bundles/azure-bundle/src/main/java/org/apache/gravitino/abs/credential/ADLSAccountKeyProvider.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+package org.apache.gravitino.abs.credential;
+
+import java.util.Map;
+import org.apache.gravitino.credential.AzureAccountKeyCredential;
+import org.apache.gravitino.credential.Credential;
+import org.apache.gravitino.credential.CredentialConstants;
+import org.apache.gravitino.credential.CredentialContext;
+import org.apache.gravitino.credential.CredentialProvider;
+import org.apache.gravitino.credential.config.ADLSCredentialConfig;
```
Comment (FANNG1) [inline bundles/azure-bundle/src/main/java/org/apache/gravitino/abs/credential/ADLSAccountKeyProvider.java:28]: rename `ADLSCredentialConfig` to `AzureCredentialConfig`?

**Comment by orenccl** [comment]
I didn't check carefully earlier.

I think we should use `ADLSAccountKeyProvider` and `ADLSAccountKeyCredential`. We can use `ADLSAccountKeyProvider` to generate `ADLSAccountKeyCredential`.

`CredentialPropertyUtils` should recognize it and convert it to Iceberg configuration like this:
```java
@VisibleForTesting static final String ICEBERG_ADLS_ACCOUNT_NAME = "adls.auth.shared-key.account.name";

@VisibleForTesting static final String ICEBERG_ADLS_ACCOUNT_KEY = "adls.auth.shared-key.account.key";
```

We could change ADLSCredentialConfig to AzureCredentialConfig, as it's more general."

File: docs/iceberg-rest-service.md [hunk 5]
```diff
@@ -175,15 +175,15 @@ Please set `gravitino.iceberg-rest.warehouse` to `gs://{bucket_name}/${prefix_na
...
+|-----------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------|----------|------------------|
+| `gravitino.iceberg-rest.io-impl`                    | The IO implementation for `FileIO` in Iceberg, use `org.apache.iceberg.azure.adlsv2.ADLSFileIO` for ADLS.                                                                                           | (none)        | Yes      | 0.8.0-incubating |
+| `gravitino.iceberg-rest.credential-provider-type`   | Supports `adls-token` and `adls-account-key`. `adls-token` generates a temporary token according to the query data path while `adls-account-key` using the storage account key to access ADLS data. | (none)        | Yes      | 0.8.0-incubating |
```
Comment (tengqm) [inline docs/iceberg-rest-service.md:181]:
```suggestion
| `gravitino.iceberg-rest.credential-provider-type`   | Supports `adls-token` and `adls-account-key`. `adls-token` generates a temporary token according to the query data path while `adls-account-key` uses a storage account key to access ADLS data. | (none)        | Yes      | 0.8.0-incubating |
```

File: api/src/main/java/org/apache/gravitino/credential/AzureAccountKeyCredential.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    return accountKey;
+  }
+
+  private void validate(String accountName, String accountKey, long expireTimeInMS) {
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(accountName), "Azure account name should not be empty.");
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(accountKey), "Azure account key should not be empty.");
+    Preconditions.checkArgument(
+        expireTimeInMS == 0, "The expire time of AzureAccountKeyCredential is not 0");
```
Comment (tengqm) [inline api/src/main/java/org/apache/gravitino/credential/AzureAccountKeyCredential.java:109]: This check doesn't make good senses to me.
Could you please double check?

**Comment by FANNG1** [comment]
> @FANNG1 I didn't check carefully earlier.
> 
> I think we should use `ADLSAccountKeyProvider` and `ADLSAccountKeyCredential`. We can use `ADLSAccountKeyProvider` to generate `ADLSAccountKeyCredential`.
> 
> ```java
> public class ADLSAccountKeyCredential implements Credential {
>   public static final String ADLS_ACCOUNT_KEY_CREDENTIAL_TYPE = "adls-account-key";
>   public static final String GRAVITINO_ADLS_STORAGE_ACCOUNT_NAME = "adls-storage-account-name";
>   public static final String GRAVITINO_ADLS_STORAGE_ACCOUNT_KEY = "adls-storage-account-key";
> ```
> 
> `CredentialPropertyUtils` should recognize it and convert it to Iceberg configuration like this:
> 
> ```java
> @VisibleForTesting static final String ICEBERG_ADLS_ACCOUNT_NAME = "adls.auth.shared-key.account.name";
> @VisibleForTesting static final String ICEBERG_ADLS_ACCOUNT_KEY = "adls.auth.shared-key.account.key";
> 
> ImmutableMap.of(
>   ADLSAccountKeyCredential.GRAVITINO_ADLS_STORAGE_ACCOUNT_NAME,
>   ICEBERG_ADLS_ACCOUNT_NAME,
>   ADLSAccountKeyCredential.GRAVITINO_ADLS_STORAGE_ACCOUNT_KEY,
>   ICEBERG_ADLS_ACCOUNT_KEY);
> 
>  if(credential instanceof ADLSAccountKeyCredential) {
>    return transformProperties(credential.credentialInfo(), icebergCredentialPropertyMap);
>  }
> ```
> 
> Iceberg use `adls.auth.shared-key.account.name`, `adls.auth.shared-key.account.key`, to determine that it will use ADLS.
> 
> We could change ADLSCredentialConfig to AzureCredentialConfig, as it's more general.
> 
> WDYT?

For Iceberg it just supports ADLS, but this credential will be used by Gravitino server too, and if supports ABS storage, it will be odd to use ADLSxxxCredential.

File: api/src/main/java/org/apache/gravitino/credential/AzureAccountKeyCredential.java (hunk 6 above)
Comment (orenccl) [reply api/src/main/java/org/apache/gravitino/credential/AzureAccountKeyCredential.java:109]: Actually can remove this line 
`Preconditions.checkArgument(expireTimeInMS == 0, "The expire time of AzureAccountKeyCredential is not 0");`

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTADLSAccountKeyIT.java (hunk 2 above)
Comment (orenccl) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTADLSAccountKeyIT.java]: fixed related class

File: docs/iceberg-rest-service.md (hunk 5 above)
Comment (orenccl) [reply docs/iceberg-rest-service.md]: Fixed. Thank you.

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
LGTM, @tengqm any other comments?

**Comment by tengqm** [comment]
> LGTM, @tengqm any other comments?

All good from my side.

**Pull request merged**